src/components/Chat/ShareModal.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  FloatingFocusManager,
+  FloatingOverlay,
+  FloatingPortal,
+  useDismiss,
+  useFloating,
+  useInteractions,
+} from '@floating-ui/react';
+import { IconCopy, IconX } from '@tabler/icons-react';
+import { IconCheck } from '@tabler/icons-react';
+import { MouseEvent, useCallback, useRef, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import { Tooltip, TooltipContent, TooltipTrigger } from '../Common/Tooltip';
+
+import { v4 as uuidv4 } from 'uuid';
+
+export enum SharingType {
+  Conversation = 'conversation',
+  ConversationFolder = 'conversations_folder',
+  Prompt = 'prompt',
+  PromptFolder = 'prompts_folder',
+}
+
+interface Entity {
+  id: string;
+  name: string;
+}
+
+interface Props {
+  entity: Entity;
+  type: SharingType;
+  isOpen: boolean;
+  onClose: () => void;
+  onShare: (shareId: string) => void;
+}
+
+export default function ShareModal({
+  entity,
+  isOpen,
+  onClose,
+  onShare,
+  type,
+}: Props) {
+  const { t } = useTranslation('sidebar');
+  const copyButtonRef = useRef<HTMLButtonElement>(null);
+  const [urlCopied, setUrlCopied] = useState(false);
+  const [urlWasCopied, setUrlWasCopied] = useState(false);
+  const shareId = uuidv4();
```

Comment:
Is it creating new uuid on every rerender?

Reply:
yes, already fixed